File: AircraftWar/src/main/scala/work/chiro/game/prop/BombProp.scala
package work.chiro.game.prop

import work.chiro.game.GlobalConfigLoader.config
import work.chiro.game.aircraft.{AbstractAircraft, HeroAircraft}
import work.chiro.game.animate.{AnimateContainer, AnimateLinearToTarget, AnimateVectorType}
import work.chiro.game.application.{ImageResourceFactory, Main}
import work.chiro.game.basic.PositionType.Position
import work.chiro.game.basic.Vec2Double
import work.chiro.game.logger
import work.chiro.game.utils.getTimeMills

class BombProp(posInit: Position, animateContainer: AnimateContainer[Vec2Double])
  extends AbstractProp(posInit, animateContainer) {
  override def handleAircrafts(enemyAircrafts: List[AbstractAircraft]) = {
    logger.info("BombSupply active!")
    enemyAircrafts.foreach(_.vanish())
  }

  override def create(position: Position) = BombProp.create(position)

  override def getImage = BombProp.getImage
}

object BombProp extends ImageResourceFactory {
  override def getImageCachedPath = "images/prop_bomb.png"

  def create(position: Position) = {
    new BombProp(
      position, new AnimateContainer[Position](List(
        new AnimateLinearToTarget(position, new Position(position.getX, config.window.height),
          AnimateVectorType.PositionLike.id, getTimeMills, 20000)
      ))
    )
  }
}

File: base/src/main/java/edu/hitsz/prop/BombProp.java
package edu.hitsz.prop;

import edu.hitsz.aircraft.AbstractAircraft;
import edu.hitsz.vector.Vec2;

import java.util.List;

/**
 * @author Chiro
 */
public class BombProp extends AbstractProp {
    public BombProp(Vec2 posInit) {
        super(posInit);
    }

    @Override
    public void handleAircrafts(List<AbstractAircraft> enemyAircrafts) {
        System.out.println("BombSupply active!");
        for (AbstractAircraft enemy : enemyAircrafts) {
            enemy.vanish();
        }
    }
}
